shared/src/main/scala/fetch.scala

```diff
@@ -268,17 +268,28 @@ object `package` {
     def run[F[_]]: FetchRunner[F] = new FetchRunner[F]
 
     private[fetch] class FetchRunner[F[_]](private val dummy: Boolean = true) extends AnyVal {
+      def apply[A](
+        fa: Fetch[F, A]
+      )(
+        implicit
+          P: Par[F],
+          C: ConcurrentEffect[F],
```

Comment:
Could be this replaced by `Monad`?

Reply:
I'm afraid is not. The effect type `F[_]` of the `DataSource` instances requires a `ConcurrentEffect[F]` and `Par[F]`. For running `F` we need `ContextShift[F]` and `Timer[F]`so these are the less restrictive implicits.

Reply:
I think we need `Sync` for `Ref.of` and the `DataSource` methods still require `ConcurrentEffect` at the moment.
So `performRun` and co need `ConcurrentEffect` as well.